lib/bulk_insert/worker.rb

```diff
@@ -64,34 +68,50 @@ def after_save(&block)
 
     def save!
       if pending?
-        sql = "INSERT #{@ignore} INTO #{@table_name} (#{@column_names}) VALUES "
-        @now = Time.now
-
-        rows = []
-        @set.each do |row|
-          values = []
-          @columns.zip(row) do |column, value|
-            value = @now if value == :__timestamp_placeholder
-
-            if ActiveRecord::VERSION::STRING >= "5.0.0"
-              value = @connection.type_cast_from_column(column, value) if column
-              values << @connection.quote(value)
-            else
-              values << @connection.quote(value, column)
-            end
-          end
-          rows << "(#{values.join(',')})"
-        end
-
-        sql << rows.join(",")
-        @connection.execute(sql)
-
+        @connection.execute(compose_insert_query)
         @after_save_callback.() if @after_save_callback
-
         @set.clear
       end
 
       self
     end
+
+    private
+
+    def compose_insert_query
+      sql = insert_sql_statement
+      @now = Time.now
+
+      rows = []
+      @set.each do |row|
+        values = []
+        @columns.zip(row) do |column, value|
+          value = @now if value == :__timestamp_placeholder
+
+          if ActiveRecord::VERSION::STRING >= "5.0.0"
+            value = @connection.type_cast_from_column(column, value) if column
+            values << @connection.quote(value)
+          else
+            values << @connection.quote(value, column)
+          end
+        end
+        rows << "(#{values.join(',')})"
+      end
+
+      sql << rows.join(",")
+      sql << on_conflict_statement
+      sql
+    end
+
+    def insert_sql_statement
+      ignore = nil
+      ignore = 'IGNORE' if (adapter_name == 'MySQL') && @ignore
+      "INSERT #{ignore} INTO #{@table_name} (#{@column_names}) VALUES "
+    end
+
+    def on_conflict_statement
+      return ' ON CONFLICT DO NOTHING' if (adapter_name == 'PostgreSQL') && @ignore
```

Comment:
Don't use an explicit `return` here -- let the string stand on its own as you did in the `insert_sql_statement` method.